Nos/Router.swift

```diff
@@ -97,13 +99,35 @@ extension Router {
                 // handle mentions. mention link will be prefixed with "@" followed by
                 // the hex format pubkey of the mentioned author
             if link.hasPrefix("@") {
-                push(try Author.findOrCreate(by: identifier, context: persistenceController.viewContext))
+                if identifier.isValidHexadecimal {
+                    push(try Author.findOrCreate(by: identifier, context: persistenceController.viewContext))
+                } else {
+                    isLoading = true
+                    let npub = await relayService
+                        .retrievePublicKeyFromUsername(identifier)?
+                        .trimmingCharacters(
+                            in: NSCharacterSet.whitespacesAndNewlines
+                        )
+                    isLoading = false
+                    if let npub, let publicKey = PublicKey.build(npub) {
+                        push(
+                            try Author.findOrCreate(
+                                by: publicKey.hex,
+                                context: persistenceController.viewContext
+                            )
+                        )
+                    } else if let url = URL(string: "mailto:\(identifier)") {
+                        await UIApplication.shared.open(url)
+                    } else {
+                        Log.debug("Couldn't open \(identifier)")
+                    }
+                }
```

Comment:
Could we refactor some of this out to maybe a private func? Seems like we have a lot of nested layers and I think it might be a little easier to read if we can move some code out.

Reply:
I refactored it in a new function named `handleNIP05Link` and also placed the line above inside a new function `handleHexadecimalPublicKey` so that the code is more clear.